modular_skyrat/modules/corporate_diplomat/code/diplomat.dm

```diff
@@ -0,0 +1,147 @@
+
+	switch(player_client.prefs.alt_job_titles[JOB_CORPORATE_DIPLOMAT])
+		if("Nanotrasen Consultant")
+			set_diplomat_type(new /datum/corporate_diplomat_role/nanotrasen_consultant, player_client, spawning)
+
+		if("Solar Federation Liaison")
+			set_diplomat_type(new /datum/corporate_diplomat_role/solfed_liaison, player_client, spawning)
+
+		if("Armadyne Representative")
+			set_diplomat_type(new /datum/corporate_diplomat_role/armadyne_representative, player_client, spawning)
+
+
+/// If someone joins roundstart, it'll set the diplomat type to be their choice.
+/// If nobody joins, then it'll be randomly chosen through `set_random_type()`
+/datum/job/corporate_diplomat/proc/set_diplomat_type(datum/corporate_diplomat_role/picked_role, client/player_client, mob/living/carbon/human/spawning)
+	// SSjob handling
+	SSjob.corporate_diplomat_type = picked_role.type
+	SSjob.name_occupations -= JOB_CORPORATE_DIPLOMAT
+	SSjob.name_occupations[picked_role.title] = src
+
+	SEND_SIGNAL(SSjob, COMSIG_CORPORATE_DIPLOMAT_ROLE_PICKED)
+
+	if(istype(spawning))
+		spawning.equipOutfit(new picked_role.outfit)
+
+	// Carrying over data to the job from the datum
+	title = picked_role.title
+	description = picked_role.description
+	department_head = picked_role.department_head.Copy()
+	supervisors = picked_role.supervisors
+	department_for_prefs = picked_role.department_for_prefs
+	departments_list |= picked_role.departments_list
+	outfit = picked_role.outfit
+	plasmaman_outfit = picked_role.plasmaman_outfit
+	display_order = picked_role.display_order
+	family_heirlooms = picked_role.family_heirlooms.Copy()
+	mail_goodies = picked_role.mail_goodies.Copy()
+
+	// Code for sticking in a new department to SSjob
+	var/type_dept = picked_role.departments_list[1]
+	var/datum/job_department/our_dept = new type_dept
+	SSjob.joinable_departments += our_dept
+	SSjob.joinable_departments_by_type[our_dept.type] = our_dept
+	sortTim(SSjob.joinable_departments_by_type, GLOBAL_PROC_REF(cmp_department_display_asc), associative = TRUE)
+	our_dept.add_job(src)
+
+	// Delete the hanging datum
+	qdel(picked_role)
+
+
+/// Selects a random type on round start if nobody took the role.
+/datum/job/corporate_diplomat/proc/set_random_type()
+	SIGNAL_HANDLER
+
+	if(SSjob.corporate_diplomat_type)
+		var/datum/corporate_diplomat_role/picked_role = new SSjob.corporate_diplomat_type
+		set_diplomat_type(new picked_role)
+		UnregisterSignal(src, COMSIG_TICKER_ROUND_STARTING)
+		return
+
+	if(!current_positions)
+		var/datum/corporate_diplomat_role/picked_role = pick(subtypesof(/datum/corporate_diplomat_role))
+		set_diplomat_type(new picked_role)
+		UnregisterSignal(src, COMSIG_TICKER_ROUND_STARTING)
+		return
+
+
+// make sure to fill ALL of these out on the subtype
+/datum/corporate_diplomat_role
+	/// The associated access level with the role
+	var/used_access
+
+	// Everything below is mirrored from /datum/job
```

Comment:
Could be nice to have the documentation mirrored too, just for ease of reference.